staging/src/k8s.io/client-go/dynamic/scheme.go

```diff
@@ -106,3 +100,24 @@ func (t unstructuredTyper) ObjectKinds(obj runtime.Object) ([]schema.GroupVersio
 func (t unstructuredTyper) Recognizes(gvk schema.GroupVersionKind) bool {
 	return true
 }
+
+// The dynamic client formerly hardcoded object marshaling and unmarshaling and would allow
+// Unstructured objects that were missing apiVersion and/or kind in requests.
```

Comment:
Based on testing, these were serialized, but then failed when used for a create, update, or apply.  so we're changing from a server-side error after the request to a client-side error just before the request, correct?

Reply:
> Based on testing, these were serialized, but then failed when used for a create, update, or apply. so we're changing from a server-side error after the request to a client-side error just before the request, correct?

Nearly correct.  CRs fail, but built-in types accept json without kind or apiVersion.